### PR TITLE
P2P: raise RuntimeError if pyarrow version is not sufficient

### PR DESCRIPTION
**Pull request by fjetter**
The test is unfortunately never triggered on CI... we don't have a job that installs pandas but not pyarrow.

### CODE REVIEW

**Review by hendrikmakait — APPROVED** [review]
[EDIT: I overlooked the CI failures.]

Thanks @fjetter!

File: distributed/shuffle/_arrow.py [hunk 1]
```diff
@@ -28,6 +28,27 @@ def check_dtype_support(meta_input: pd.DataFrame) -> None:
...
+    # First version to introduce Table.sort_by
+    minversion = "7.0.0"
+    try:
+        import pyarrow as pa
+        from packaging.version import parse
+
+        if parse(pa.__version__) >= parse(minversion):
+            return
+        suffix = f" but only found {pa.__version__}"
+    finally:
```
Comment (hendrikmakait) [inline distributed/shuffle/_arrow.py:48]: This is triggered even on the early return above. This logic needs slight fixing to only trigger on bad versions.

**Review by hendrikmakait — CHANGES_REQUESTED** [review]

File: distributed/shuffle/_arrow.py (hunk 1 above)
Comment (fjetter) [reply distributed/shuffle/_arrow.py:48]: oh, right :D

File: distributed/shuffle/_arrow.py [hunk 2]
```diff
@@ -28,6 +30,27 @@ def check_dtype_support(meta_input: pd.DataFrame) -> None:
...
+    Raises a RuntimeError in case pyarrow is not installed or installed version
+    is not recent enough.
+    """
+    suffix = ""
+    # First version to introduce Table.sort_by
+    minversion = "7.0.0"
+    try:
+        import pyarrow as pa
+    except ImportError:
+        raise RuntimeError(f"P2P shuffling requires pyarrow>={minversion}" + suffix)
```
Comment (hendrikmakait) [inline distributed/shuffle/_arrow.py:46]:
```suggestion
    # First version to introduce Table.sort_by
    minversion = "7.0.0"
    try:
        import pyarrow as pa
    except ImportError:
        raise RuntimeError(f"P2P shuffling requires pyarrow>={minversion}")
```

**Pull request merged**